cloudinit/sources/DataSourceAzure.py

```diff
@@ -1355,6 +1385,15 @@ def region(self):
         return self.metadata.get('imds', {}).get('compute', {}).get('location')
 
 
+def _key_is_openssh_formatted(key):
+    """
+    Validate whether or not the key is OpenSSH-formatted. This
+    is done by looking for a prefix of "ssh-".
```

Comment:
This should suffice at the moment as the keys in OpenSSH format are almost all entirely "ssh-rsa" (and this is [what is documented](https://docs.microsoft.com/en-us/rest/api/compute/virtualmachines/createorupdate#sshpublickey)). We should be safe to move forward with this.

Reply:
> I don't have a physical Yubikey or similar to check with, but my reading of https://github.com/openssh/openssh-portable/blob/master/sshkey.c also suggests that "SK" keys generated using one will also not match an `ssh-` prefix.

See the reference to sshkey.c in cloudinit/ssh_util.py.  It seems that it might make more sense to pass it to AuthKeyLineParser:

```python
import ssh_util


def _key_is_openssh_formatted(key):
    parser = ssh_util.AuthKeyLineParser()
    try:
        akl = parser.parse(key)
    except TypeError:
        return False

    return akl.keytype is not None
```

Reply:
That's a good idea, thanks! I made these code changes!